### PR TITLE
269 capacity

### LINKED ISSUE #269: Funding spaces: On my roster after uploading, I can see my capacity and enrollment for each funding space.
Screen Shot 2020-09-04 at 8 45 41 AM

For the current reporting period, capacity/utilization for each space type a user's organization is allocated. 

Styling: 
font-size: 16px;
color: darker-gray (#565C65);
margin-right: 24px;
16px between lines if spaces span two lines. 

Screen Shot 2020-09-10 at 2 06 54 PM

If I have too many enrollments for a given space type, show this error:
Screen Shot 2020-09-18 at 8 31 40 AM

### LINKED ISSUE #335: Redirect non-authenticated users to the homepage
In hedwig, we redirect user to the homepage where they can login. Do we want to replicate this for Fawkes?

### LINKED ISSUE #340: Child record should open to child identifiers, not child info

### LINKED ISSUE #346: Trying to withdraw a child results in a blank screen

### PR DESCRIPTION
**Pull request by jlhogan**
closes #269 - adds capacity utilization to roster headers
closes #340 - updates initial active tab in EditRecord to child identifiers
closes #346 - fixes Withdraw modal (RIDER: Creates modal component in component lib https://github.com/ctoec/component-library/pull/69 and uses this for both Withdraw and Delete)
closes #335 - adds unauthneticated redirect for authenticated routes 

OPEN QUESTION: when something in a modal fails, should:
1) alert is displayed in the modal & modal kept open 
2) alert is displayed on main page & modal is closed
?

### CODE REVIEW

**Comment by RyanHansz** [comment]
@jlhogan any scenarios of something that may fail in a modal?

File: client/src/containers/App/NotSignedInRedirect.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export const NotSignedInRedirect: React.FC = ({ children }) => {
+  const history = useHistory();
+  const { loading, user } = useContext(UserContext);
+
+  if (loading) {
+    return <></>;
+  } else {
+    if (!user) {
+      history.push('/');
```
Comment (mmazanec22) [inline client/src/containers/App/NotSignedInRedirect.tsx:13]: We could use the redirect component from react router instead: https://reactrouter.com/web/api/Redirect

**Review by mmazanec22 — APPROVED** [review]

File: client/src/containers/Roster/RosterSectionHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+export const RosterSectionHeader: React.FC<RosterSectionHeaderProps> = ({
+  ageGroup,
+  children,
+}) => {
+  const fundingSpaceCounts: {
+    fundingSpace: FundingSpace;
+    count: number;
+  }[] = [];
+  children.reduce((acc, child) => {
```
Comment (bamader) [inline client/src/containers/Roster/RosterSectionHeader.tsx:19]: Nit nit nit: Can we use a more descriptive parameter name than `acc`? My guess is this means accumulator, but if I didn't know the function already a name might clarify that for me.

File: client/src/containers/Roster/RosterSectionHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { Child, AgeGroup, FundingSpace } from '../../shared/models';
+import { getCurrentFunding } from '../../utils/models';
+import pluralize from 'pluralize';
+
+type RosterSectionHeaderProps = {
+  ageGroup: AgeGroup;
```
Comment (bamader) [inline client/src/containers/Roster/RosterSectionHeader.tsx:7]: Is a roster made up of children only in one age group? I thought filtering by age was something to be added later, and that a roster could hold children of multiple age groups?

File: client/src/utils/models/getCurrentEnrollment.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Child } from '../../shared/models';
+import moment from 'moment';
+
+export const getCurrentEnrollment = (child: Child) => {
+  const today = moment.utc();
+  return (child.enrollments || []).find(
+    (enrollment) => !enrollment.exit || enrollment.exit.isSameOrAfter(today)
```
Comment (bamader) [inline client/src/utils/models/getCurrentEnrollment.ts:7]: Great refactor!

File: client/src/utils/models/getCurrentFunding.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { Child } from '../../shared/models';
+import moment from 'moment';
+import { getCurrentEnrollment } from './getCurrentEnrollment';
+
+export const getCurrentFunding = (child: Child) => {
```
Comment (bamader) [inline client/src/utils/models/getCurrentFunding.ts:5]: Nit: I know there are places where the application needs to determine both the current funding and the current enrollment. Could we define an overload signature of this function (or even just have an optional second parameter) where if a current enrollment is passed in, it doesn't have to look up the enrollment? I know it's probably negligible savings, but if we're refactoring and cleaning I thought I'd put it out there.

**Review by bamader — APPROVED** [review]
LGTM!

File: client/src/containers/EditRecord/DeleteRecord/index.tsx [hunk 6]
```diff
@@ -19,50 +21,70 @@ export const DeleteRecord: React.FC<DeleteProps> = ({ child, toggleOpen }) => {
...
-      .then(() =>
+      .then(() => {
+        toggleIsOpen();
         history.push('/roster', {
           alerts: [
             {
               type: 'success',
               heading: 'Record deleted',
-              text: `${child?.firstName} ${child?.lastName}'s record was delete from your roster.`,
+              text: `${child?.firstName} ${child?.lastName}'s record was deleted from your roster.`,
```
Comment (JordanGuinn) [inline client/src/containers/EditRecord/DeleteRecord/index.tsx:31]: heh

File: client/src/containers/EditRecord/DeleteRecord/index.tsx [hunk 7]
```diff
@@ -19,50 +21,70 @@ export const DeleteRecord: React.FC<DeleteProps> = ({ child, toggleOpen }) => {
...
+        text="Delete record"
+        className="margin-right-0"
+      />
+      <Modal
+        isOpen={isOpen}
+        toggleOpen={toggleIsOpen}
+        header={
+          <h2>
+            Do you want to delete the enrollment for{' '}
+            {`${child.firstName} ${child.lastName}`}
```
Comment (JordanGuinn) [inline client/src/containers/EditRecord/DeleteRecord/index.tsx:58]: The nittiest possible nit I think I've ever made - should there be a question mark here?

**Review by JordanGuinn — APPROVED** [review]
LGTM 👍🏽👍🏽👍🏽👍🏽👍🏽

File: client/src/containers/Roster/RosterSectionHeader.tsx (hunk 3 above)
Comment (jlhogan) [reply client/src/containers/Roster/RosterSectionHeader.tsx:7]: a roster _section_ is :) i can add a comment here to make more explicit

Comment (bamader) [reply client/src/containers/Roster/RosterSectionHeader.tsx:7]: Oooooh I get it. Having actually worked a bit with the code in the roster section for a different ticket, that now makes perfect sense.

File: client/src/utils/models/getCurrentFunding.ts (hunk 5 above)
Comment (jlhogan) [reply client/src/utils/models/getCurrentFunding.ts:5]: when someone needs that functionality, the absolutely should add it to the util :)

File: client/src/containers/EditRecord/DeleteRecord/index.tsx (hunk 7 above)
Comment (jlhogan) [reply client/src/containers/EditRecord/DeleteRecord/index.tsx:58]: the actual correction was to remove the other `?` -- you can't attempt to delete a child that doesn't exist :)

**Pull request merged**